examples/calculations/submit_geomopt.py

```diff
@@ -0,0 +1,188 @@
+"""Example code for submitting single point calculation"""
+
+from pathlib import Path
+from typing import Optional, Union
+
+from ase.build import bulk
+
+# from ase.io import read
+import ase.io
+import click
+
+from aiida.common import NotExistent
+from aiida.engine import run_get_node
+from aiida.orm import Bool, Dict, Float, Int, Str, StructureData, load_code, load_node
+from aiida.plugins import CalculationFactory
+
+from aiida_mlip.data.model import ModelData
+
+
+def load_model(url_or_path: Optional[Union[str, Path]], architecture: str) -> ModelData:
+    """
+    Load a model from a given string.
+
+    If the string represents a file path, the model will be loaded from that path.
+    Otherwise, the model will be downloaded from the specified location.
+
+    Parameters
+    ----------
+    url_or_path : Optional[Union[str, Path]]
+        The string representing either a file path or a URL for downloading the model.
+    architecture : str
+        The architecture of the model.
+
+    Returns
+    -------
+    ModelData
+        The loaded model.
+    """
+    if url_or_path is None:
+        model = None
+    elif (file_path := Path(url_or_path)).is_file():
+        model = ModelData.local_file(file_path, architecture=architecture)
+    else:
+        model = ModelData.download(url_or_path, architecture=architecture)
+    return model
```

Comment:
```suggestion
def load_model(model: Optional[Union[str, Path]], architecture: str) -> ModelData:
    """
    Load a model from a file path or URL.

    If the string represents a file path, the model will be loaded from that path.
    Otherwise, the model will be downloaded from the specified location.

    Parameters
    ----------
    model : Optional[Union[str, Path]]
        Model file path or a URL for downloading the model.
    architecture : str
        The architecture of the model.

    Returns
    -------
    ModelData
        The loaded model.
    """
    if model is None:
        loaded_model = None
    elif (file_path := Path(url_or_path)).is_file():
        loaded_model = ModelData.local_file(file_path, architecture=architecture)
    else:
        loaded_model = ModelData.download(url_or_path, architecture=architecture)
    return loaded_model
```
I know you've already changed this, but I wonder if something like `model` is more appropriate, similar to `struct` in `load_structure` being a `str` or `Path`, so it's clear what is actually represents.